### PR TITLE
Pulling lockers now slow you down.

### PR DESCRIPTION
**Pull request by kriskog**
Pulling lockers now slow you down.

## About The Pull Request
Lockers now gives you slowdown when dragging. No more "put body in heavy locker" to get around the dragging slowdown, and penalizes dragging a locker around to block projectiles. The slowdown does not apply to crates (because cargo) or bodybags.

## Why It's Good For The Game
Lockers are bulky, makes them behave that way. Less of gamers dragging around an obstacle that you can toggle density of at will.

## Changelog
:cl: Skoglol
balance: Pulling a locker now slows you down.
/:cl:

### CODE REVIEW

**Comment by nemvar** [comment]
Time to drag crates around.

File: code/modules/mob/living/living_movement.dm [hunk 1]
```diff
@@ -42,12 +42,18 @@
...
+			var/mob/living/L = pulling
+			if(L.drag_slowdown && !(L.mobility_flags & MOBILITY_STAND) && !L.buckled && grab_state < GRAB_AGGRESSIVE)
+				add_movespeed_modifier(MOVESPEED_ID_BULKY_DRAGGING, multiplicative_slowdown = PULL_BULKY_SLOWDOWN)
+				return
+		if(isstructure(pulling))
+			var/obj/structure/S = pulling
+			if(S.drag_slowdown)
+				add_movespeed_modifier(MOVESPEED_ID_BULKY_DRAGGING, multiplicative_slowdown = PULL_BULKY_SLOWDOWN)
+				return
+	remove_movespeed_modifier(MOVESPEED_ID_BULKY_DRAGGING)
```
Comment (nemvar) [inline code/modules/mob/living/living_movement.dm]: This entire proc has bad structure. Try again.

**Comment by Dawson1917** [comment]
> 
> 
> I'm pretty sure this was an intentional interaction

"It'll be left in for now and revisited if it becomes too much of a meta after this PR" or something of that degree was said. Doesn't seem like it's been too bad to me. Fireman carrying is just slow af, both to pick someone up and moving with them.

**Comment by kriskog** [comment]
The lockers act as a barrier you can put up instantly and drag behind you, basically movable cover. Crates can not be made non dense, and are fine in that regard.

**Comment by kriskog** [comment]
Crates are harder to navigate around, since you gotta climb over them. You can't just open a crate to turn around in a pinch, like you can with the locker. If you are behind a crate, getting in front of it to attack is much harder than a locker.
Also cargo sells crates, I would hate to make cargo even less fun.

**Comment by kriskog** [comment]
> 
>  Doesn't seem like it's been too bad to me. Fireman carrying is just slow af, both to pick someone up and moving with them.

Note that this isn't just targeting bodies putting bodies in the locker, but also the mobile barricade. There is a reason you currently see full black spraycanned people (sec and antags alike) running around with lockers and no intention to kidnap anyone.

**Comment by Ratvar** [comment]
This PR wouldn't have so many downthumbs if it wasn't meta : -          )

**Comment by imsxz** [comment]
no

**Comment by ATH1909** [comment]
> > Doesn't seem like it's been too bad to me. Fireman carrying is just slow af, both to pick someone up and moving with them.
> 
> Note that this isn't just targeting putting bodies in the locker, but also the mobile barricade. There is a reason you currently see full black spraycanned people (sec and antags alike) running around with lockers and no intention to kidnap anyone.
> Medical personell (and antags) can still use bodybags or roller beds for bodies, its just not quite as useful as the mighty locker.
> 
> > Ruled a non-issue, head maintainer said he wanted to see how the meta shakes out, see #45667
> 
> The meta shook out, it is now a gamer must have.

While do see the combat value of a locker as a disabler shield, I have seen NO ONE on BAGIL (the most powergame-y of the four servers) casually walk around while dragging a locker.

**Comment by kriskog** [comment]
If you think bagil is powergamey, come by Terry some day. This is far from a rare sight.
Heck, I am guilty of doing this myself some times.

**Comment by wesoda25** [comment]
its so easy lol just pull a locker behind you whille running and you are immune to disabler shots. just because its the meta doesnt mean it should be removed

File: code/modules/mob/living/living_movement.dm (hunk 1 above)
Comment (kriskog) [reply code/modules/mob/living/living_movement.dm]: better?

**Comment by SkeletalElite** [comment]
Welp time to use a rolling chair instead of a locker

**Comment by kriskog** [comment]
A rolling chair has wheels and is not a barricade on demand. I am fine with that.

**Comment by SkeletalElite** [comment]
> A rolling chair has wheels and is not a barricade on demand. I am fine with that.

rolling chair with a body in it will still block projectiles since the projectiles hit the body.

**Comment by kriskog** [comment]
> 
> 
> > A rolling chair has wheels and is not a barricade on demand. I am fine with that.
> 
> rolling chair with a body in it will still block projectiles since the projectiles hit the body.

This is true, but it will be less practical.
 - You can't pass the chair without unbuckling, then rebuckling
 - You can't click it to activate the "steal stunbaton" mode
 - Rolling a body around is much less legitimate and a cause for concern
 - Getting a roller chair and a body requires more work than just ctrl clicking one of the countless lockers as you run past
- Since it is always dense, you will keep dropping that chair fairly often

**Comment by actioninja** [comment]
Thought about this a bit and I think this will be a positive change in the long run. I never have been a big fan of excessively "gamey" stuff.  While using lockers to move bodies and the like quicker was a semi-intended interaction, some of things people are doing like using lockers as mobile cover and baton traps is stupid.

Before it's brought up I am aware this is a nerf to being able to easily move multiple bodies, grab a body bag if you really need that functionality.

File: code/game/objects/structures.dm [hunk 2]
```diff
@@ -3,12 +3,14 @@
...
 	interaction_flags_atom = INTERACT_ATOM_ATTACK_HAND | INTERACT_ATOM_UI_INTERACT
+	layer = BELOW_OBJ_LAYER
+	
 	var/climb_time = 20
 	var/climb_stun = 20
 	var/climbable = FALSE
 	var/mob/living/structureclimber
 	var/broken = 0 //similar to machinery's stat BROKEN
-	layer = BELOW_OBJ_LAYER
+	var/drag_slowdown //If the structure causes slowdown when dragged
```
Comment (actioninja) [inline code/game/objects/structures.dm]: identical var defines on disparate typepaths is gross.

**Review by actioninja — CHANGES_REQUESTED** [review]
stick the slowdown var on obj and make it not a boolean but a value that's added as slowdown
keep the old mob drag_slowdown it's intentionally whether or not the mob dragging receives slowdown not whether the draggee causes slowdown.

File: code/modules/mob/living/living_defines.dm [hunk 3]
```diff
@@ -117,4 +117,4 @@
 	var/list/diseases = list() // list of all diseases in a mob
 	var/list/disease_resistances = list()
 
-	var/drag_slowdown = TRUE //Whether the mob is slowed down when dragging another prone mob
```
[the remark this answers is not in the document]
Comment (kriskog) [reply code/modules/mob/living/living_defines.dm]: Reverted, renamed to be more descriptive.

**Comment by kriskog** [comment]
Moved the structure var to obj, made it multiplicative so it can be adjusted per obj. In future I would argue this should exist in living as well to make pulling a dead cat and a dead body different, but that's just a bunch of work.

**Comment by kriskog** [comment]
This is still a bit wrong, I need to fix a few things.

**Comment by kriskog** [comment]
Okay, this works as intended.

**Comment by redmoogle** [comment]
this is good gameplay and discourages cheap tatics

**Comment by Timonkeyn** [comment]
time to drag roller beds around

**Comment by actioninja** [comment]
You can fold up rollerbeds carrying them with you is an entirely intended functionality

**Comment by actioninja** [comment]
The code still feels kind of shitty but at the same time I can't really think of a better way of doing it

File: code/modules/mob/living/living_movement.dm (hunk 1 above)
Comment (nemvar) [reply code/modules/mob/living/living_movement.dm]: yes

**Pull request merged**